src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -56,7 +63,17 @@ public override T[] Rent(int minimumLength)
 
             // Gettings here means we have too big of a request OR all the buckets from 
             // index through _buckets.Length are taken so we need to allocate a buffer on-demand.
-            return new T[Utilities.GetMaxSizeForBucket(index)];
+            int maxLength = Utilities.GetMaxSizeForBucket(_buckets.Length);
```

Comment:
```maxLength``` appears to only be used in the logging code.  Shouldn't this be in the if block under ```log.IsEnabled()```?